Sources/MapboxMaps/Gestures/GestureHandlers/PinchGestureHandler.swift

```diff
@@ -10,151 +12,100 @@ internal final class PinchGestureHandler: GestureHandler, PinchGestureHandlerPro
     /// Whether pinch gesture can rotate map or not
     internal var rotateEnabled: Bool = true
 
-    /// The midpoint of the touches in the gesture's view when the gesture began
-    private var initialPinchMidpoint: CGPoint?
+    /// Whether pinch gesture can zoom map or not
+    internal var zoomEnabled: Bool = true
 
-    /// The angle from touch location 0 to touch location 1 when the gesture began or unpaused
-    private var initialPinchAngle: CGFloat?
+    /// Whether pinch gesture can pan map or not
+    internal var panEnabled: Bool = true
 
-    /// The camera center when the gesture began or unpaused
-    private var initialCenter: CLLocationCoordinate2D?
+    /// The behavior for the current gesture, based on the initial state of the \*Enabled flags.
+    private var pinchBehavior: PinchBehavior?
 
-    /// The camera zoom when the gesture began
-    private var initialZoom: CGFloat?
-
-    /// The camera bearing when the gesture began or unpaused
-    private var initialBearing: CLLocationDirection?
-
-    /// The rotateEnabled setting when the gesture began
-    private var initialRotateEnabled: Bool?
-
-    private var gestureBegan = false
+    private var invokedGestureBegan = false
 
     private let mapboxMap: MapboxMapProtocol
 
+    private let pinchBehaviorProvider: PinchBehaviorProviderProtocol
+
     /// Initialize the handler which creates the panGestureRecognizer and adds to the view
     internal init(gestureRecognizer: UIPinchGestureRecognizer,
-                  mapboxMap: MapboxMapProtocol) {
+                  mapboxMap: MapboxMapProtocol,
+                  pinchBehaviorProvider: PinchBehaviorProviderProtocol) {
         self.mapboxMap = mapboxMap
+        self.pinchBehaviorProvider = pinchBehaviorProvider
         super.init(gestureRecognizer: gestureRecognizer)
         gestureRecognizer.addTarget(self, action: #selector(handleGesture(_:)))
     }
 
-    // swiftlint:disable:next cyclomatic_complexity
     @objc private func handleGesture(_ gestureRecognizer: UIPinchGestureRecognizer) {
         guard let view = gestureRecognizer.view else {
             return
         }
-        let pinchMidpoint = gestureRecognizer.location(in: view)
-
         switch gestureRecognizer.state {
         case .began:
+            // UIPinchGestureRecognizer sometimes begins with 1 touch.
+            // If that happens, we ignore it here, but will start handling it
+            // in .changed if the number of touches increases to 2.
```

Comment:
@macdrevx Any chance you can recall where from that come from?

Reply:
I don't think I ever reproduced it. IIRC it was causing a crash though (maybe in nav?) which is why this was added. I would trace back through the git blame to see if there are details in the PR where this was introduced.